src/interchange/src/avro.rs

```diff
@@ -1482,7 +1550,7 @@ impl DebeziumDeduplicationState {
     ) -> bool {
         let (pos, row) = match row {
             RowCoordinates::MySql { pos, row } => (pos, row),
-            RowCoordinates::Postgres { lsn } => (lsn, 0),
+            RowCoordinates::Postgres { lsn, total_order } => (lsn, total_order.unwrap_or(0)),
```

Comment:
I believe we can assert that `total_order` is always defined? Is there a case where a Postgres row does not contain a `total_order` field?

Reply:
I believe it's possible to turn off transaction metadata entirely, in which case we would still have `None` here.

Reply:
so I am pretty sure that this will do a lexical comparison and accept rows where the lsn went up but the total_order went down -- not sure if that's what we want to do?

Reply:
That's true, it will do that. My understanding of the Debezium code is that total_order is per transaction, so that is the correct behavior.

Reply:
so I'm not totally sure what the mapping is between lsns and transactions but I've observed at least one pair of rows where the lsn went up, and the total order went up as well (the txn id didn't change between them)

```
{"before": null, "after": {"Value": {"i_id": 65, "i_im_id": {"int": 1139}, "i_name": {"string": "1IP8MZQvubgUZYMyCS      "}, "i_price": {"bytes": "\bm"}, "i_data": {"string": "lB5fTyp6GPWTpKLzu1ncYOl@g6KrJt                    "}}}, "source": {"version": "1.2.5.Final", "connector": "postgresql", "name": "debezium", "ts_ms": 1607060299198, "snapshot": {"string": "false"}, "db": "postgres", "schema": "tpcch", "table": "item", "txId": {"long": 644}, "lsn": {"long": 1403798720}, "xmin": null}, "op": "c", "ts_ms": {"long": 1607060469161}, "transaction": {"ConnectDefault": {"id": "644", "total_order": 65, "data_collection_order": 65}}}
{"before": null, "after": {"Value": {"i_id": 66, "i_im_id": {"int": 1854}, "i_name": {"string": "lZDkDa?Fo5gKdgROFO6F0o  "}, "i_price": {"bytes": "\u0001*"}, "i_data": {"string": "KUCOfUSl7J@uLWXdMU7r4JW42sr080                    "}}}, "source": {"version": "1.2.5.Final", "connector": "postgresql", "name": "debezium", "ts_ms": 1607060299198, "snapshot": {"string": "false"}, "db": "postgres", "schema": "tpcch", "table": "item", "txId": {"long": 644}, "lsn": {"long": 1403805144}, "xmin": null}, "op": "c", "ts_ms": {"long": 1607060469161}, "transaction": {"ConnectDefault": {"id": "644", "total_order": 66, "data_collection_order": 66}}}
```

my only question is that we currently accept the second row regardless of what it said in the `txId` field or what it said in the `total_order` field as long as the second row's `lsn` was greater than the first rows `lsn` -- I can't tell if that's correct though it totally might be

Reply:
I think it's the correct behavior -- Debezium reads the WAL in order, so the "true" order of Debezium messages should correspond to increasing LSNs.